apps/qubit/modules/object/actions/exportAction.class.php

```diff
@@ -69,54 +157,142 @@ protected function addField($name)
+      case 'includeDigitalObjects':
+        if ($this->digitalObjectsAvailable)
         {
-          $choices = array(
-            $this->objectType => sfConfig::get('app_ui_label_'.strtolower($this->objectType))
-          );
+          switch ($this->objectType)
+          {
+            case 'informationObject':
+              $this->helpMessage['20'] = __('Note: Digital objects with restricted access or copyright will not be exported.') . ' ' . __('It is not possible to select both digital objects and descendants for export at the same time. Digital objects can only be exported for records that are on the clipboard.');
+
+              break;
+
+            case 'actor':
+              $this->helpMessage['20'] = __('Note: Digital objects with restricted access or copyright will not be exported.');
+
+              break;
+          }
+          $this->form->setWidget('includeDigitalObjects', new sfWidgetFormInputCheckbox(array('label' => __('Include digital objects'))));
+          $this->form->setDefault('includeDigitalObjects', true);
         }
         else
         {
-          $choices = array(
-            'informationObject' => sfConfig::get('app_ui_label_informationobject'),
-            'actor' => sfConfig::get('app_ui_label_actor'),
-            'repository' => sfConfig::get('app_ui_label_repository')
-          );
+          $this->form->setWidget('includeDigitalObjects', new sfWidgetFormInputHidden(array('label' => false)));
         }
 
-        $this->form->setValidator('objectType', new sfValidatorString(array('required' => true)));
-        $this->form->setWidget('objectType', new sfWidgetFormSelect(array('choices' => $choices)));
-
         break;
 
-      case 'format':
-        $choices = array(
-          'csv' => $this->context->i18n->__('CSV'),
-          'xml' => $this->context->i18n->__('XML')
-        );
-
-        $this->form->setDefault('format', $this->format);
-        $this->form->setValidator('format', new sfValidatorString(array('required' => true)));
-        $this->form->setWidget('format', new sfWidgetFormSelect(array('choices' => $choices)));
+      // Enable field includeDrafts if:
+      // options enabled
+      // and, user is authenticated
+      case 'includeDrafts':
+        if ($this->showOptions && $this->context->user->isAuthenticated())
+        {
+          $this->form->setWidget('includeDrafts', new sfWidgetFormInputCheckbox(array('label' => __('Include draft records'))));
+          $this->helpMessage['30'] = __('Choosing "Include draft records" will include those marked with a Draft publication status in the export. Note: if you do NOT choose this option, any descendants of a draft record will also be excluded, even if they are published.');
+          $this->form->setDefault('includeDrafts', true);
+        }
+        else
+        {
+          $this->form->setWidget('includeDrafts', new sfWidgetFormInputHidden(array('label' => false)));
+        }
```

Comment:
Could do that Radda - but was trying to avoid conditional logic in the templates where possible. No problem to remove the hiddens and go with if/then in template if preferred?

Reply:
Didn't fully realise that isset() is ok to use here - thought that non-defined fields might impact any validators. Have updated in form and action class. Much cleaner.